addons/maintainence-agent/maintenance_mode_controller.go

```diff
@@ -83,37 +72,63 @@ func (r *MaintenanceModeReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 
 	if !mmode.GetDeletionTimestamp().IsZero() || len(mmode.Spec.Modes) == 0 {
-		result, err := r.startMaintenanceActions(ctx, mmode, actionableCephClusters, false)
+		if !utils.ContainsString(mmode.GetFinalizers(), MaintenanceModeFinalizer) {
+			return ctrl.Result{}, nil
+		}
+
+		result, err := r.startMaintenanceActions(ctx, &mmode, actionableCephClusters, false)
 		if err != nil {
 			return result, err
 		}
-		mmode.Finalizers = utils.RemoveString(mmode.Finalizers, MaintenanceModeFinalizer)
 
+		mmode.Finalizers = utils.RemoveString(mmode.Finalizers, MaintenanceModeFinalizer)
 		if err := r.SpokeClient.Update(ctx, &mmode); err != nil {
 			klog.Error(err, "failed to remove finalizer from MaintenanceMode ", err)
 			return ctrl.Result{}, err
 		}
-		klog.Info("MaintenanceMode deleted, skipping reconciliation")
+
+		klog.Info("MaintenanceMode disabled")
 		return ctrl.Result{}, nil
-	} else {
-		_, err := r.startMaintenanceActions(ctx, mmode, actionableCephClusters, true)
-		if err != nil {
+	}
+
```

Comment:
If the above if fails, then both booleans were false, IOW the MMode is not deleted and the length of modes is not 0, hence process the request.